swabbie-orca/src/main/kotlin/com/netflix/spinnaker/swabbie/orca/OrcaTaskMonitoringAgent.kt

```diff
@@ -0,0 +1,214 @@
+  private val orcaService: OrcaService,
+  private val applicationEventPublisher: ApplicationEventPublisher,
+  private val retrySupport: RetrySupport,
+  private val lockingService: Optional<LockingService>
+) : DiscoveryActivated, MetricsSupport(registry) {
+
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+  private val worker: Scheduler.Worker = Schedulers.io().createWorker()
+
+  private val timeoutMillis: Long = 5000
+  private val maxAttempts: Int = 3
+
+  override val onDiscoveryUpCallback: (event: RemoteStatusChangedEvent) -> Unit
+    get() = {
+      worker.schedulePeriodically({
+       withLocking(javaClass.simpleName) { monitorOrcaTasks() }
+      }, getAgentDelay(), getAgentFrequency(), TimeUnit.SECONDS)
+    }
+
+  override val onDiscoveryDownCallback: (event: RemoteStatusChangedEvent) -> Unit
+    get() = { stop() }
+
+  @PostConstruct
+  private fun init() {
+    log.info("Initializing agent ${javaClass.simpleName}")
+    registry.gauge(lastRunAgeId.withTag("agentName", javaClass.simpleName), this) {
+      Duration
+        .between(it.getLastAgentRun(), clock.instant())
+        .toMillis().toDouble()
+    }
+  }
+
+  @PreDestroy
+  private fun stop() {
+    log.info("Stopping agent ${javaClass.simpleName}")
+    if (!worker.isUnsubscribed) {
+      worker.unsubscribe()
+    }
+  }
+
+  private fun withLocking(
+    agentName: String,
+    callback: () -> Unit
+  ) {
+    if (lockingService.isPresent) {
+      val normalizedLockName = (agentName)
+        .replace(":", ".")
+        .toLowerCase()
+      val lockOptions = LockManager.LockOptions()
+        .withLockName(normalizedLockName)
+        .withMaximumLockDuration(lockingService.get().swabbieMaxLockDuration)
+      lockingService.get().acquireLock(lockOptions) {
+        callback.invoke()
+      }
+    } else {
+      log.warn("***Locking not ENABLED***")
+      callback.invoke()
+    }
+  }
+
+  private fun monitorOrcaTasks() {
+    initialize()
+
+    val inProgressTasks = taskTrackingRepository.getInProgress()
+    if (inProgressTasks.isEmpty()) {
+      log.debug("No active orca tasks to monitor from ${javaClass.simpleName}.")
+    }
+
+    inProgressTasks
+      .forEach { taskId ->
+        val response = getTask(taskId)
+        if (response.status.isComplete()) {
+          val taskInfo = taskTrackingRepository.getTaskDetail(taskId)
+          if (taskInfo == null) {
+            log.error("Detail for task $taskId was not found in repository. Something went wrong!")
+            return
+          }
+
+          when {
+            response.status.isSuccess() -> {
+              log.debug("Orca task $taskId succeeded. Task complete info: $taskInfo")
+              publishEvent(taskInfo)
+              taskTrackingRepository.setSucceeded(taskId)
+            }
+            response.status.isFailure() -> {
+              log.error("Orca task $taskId did not complete. Status: ${response.status}. Task complete info: $taskInfo")
```

Comment:
We need to go through all of swabbie and get it logging structured logs, but it's not too late for new code. Use `StructuredArguments.kv` for things like taskId, status, etc so we can index and search through the logs better.

Reply:
I'll use this for new things and clean up as I go. Not gonna add more sweeping changes to this PR